types/src/on_chain_config/consensus_config.rs

```diff
@@ -44,60 +189,64 @@ impl OnChainConsensusConfig {
     // to this max size.
     pub fn max_failed_authors_to_store(&self) -> usize {
         match &self {
-            OnChainConsensusConfig::V1(config)
-            | OnChainConsensusConfig::V2(config)
-            | OnChainConsensusConfig::V3(ConsensusConfigV1Ext { main: config, .. }) => {
+            OnChainConsensusConfig::V1(config) | OnChainConsensusConfig::V2(config) => {
                 config.max_failed_authors_to_store
             },
-            _ => unimplemented!("method not supported"),
+            OnChainConsensusConfig::V3 { alg, .. } => alg.max_failed_authors_to_store(),
         }
     }
 
     // Type and configuration used for proposer election.
     pub fn proposer_election_type(&self) -> &ProposerElectionType {
         match &self {
-            OnChainConsensusConfig::V1(config)
-            | OnChainConsensusConfig::V2(config)
-            | OnChainConsensusConfig::V3(ConsensusConfigV1Ext { main: config, .. }) => {
+            OnChainConsensusConfig::V1(config) | OnChainConsensusConfig::V2(config) => {
                 &config.proposer_election_type
             },
-            _ => unimplemented!("method not supported"),
+            OnChainConsensusConfig::V3 { alg, .. } => alg.proposer_election_type(),
         }
     }
 
     pub fn quorum_store_enabled(&self) -> bool {
         match &self {
             OnChainConsensusConfig::V1(_config) => false,
-            OnChainConsensusConfig::V2(_) | OnChainConsensusConfig::V3(_) => true,
-            OnChainConsensusConfig::DagV1(_) => false,
+            OnChainConsensusConfig::V2(_) => true,
+            OnChainConsensusConfig::V3 { alg, .. } => alg.quorum_store_enabled(),
```

Comment:
Hmm, so we are turning off quorum store in V3 and we need to turn it back on explicitly? Why did we choose to make it explicit now?

Reply:
fixed